scripts/source-maintenance.sh

```diff
@@ -153,6 +153,9 @@ applyPluginsTo ()
 # succeeds if all MakeNamedErrorDetail() names are unique
 checkMakeNamedErrorDetails ()
 {
+    # old git versions do not support --only-matching, skip this check
+    git grep --only-matching . . >/dev/null 2>&1 || return 0
```

Comment:
We should not skip this check just because it does not work. It is wrong to admit MakeNamedErrorDetail bugs into the repository while pretending to check that there are no such bugs.

If really needed, we can allow the person running source-maintenance.sh to manually disable some required checks via command-line options, but I am not sure the complexity is worth it -- that person can already remove checks by editing the script. The end goal here is to (automatically) format every pull request anyway, removing the vast majority of these problems.

Reply:
It is unfortunate that we did not catch the --only-matching problem when the MakeNamedErrorDetail changes went in. Right now we need to disable this check on master server due to old git version 2.17. Manual runs with newer git supporting that option still run fine and can detect MakeNamedErrorDetail bugs in local tests we do.

Until we get an upgrade of master and/or git on master we need to skip this test. If you want to make it run in a robust way feel free to do a new PR re-writing the scripts test without that option.

Reply:
> Right now we need to disable this check on master server due to old git version 2.17
...
> Until we get an upgrade of master and/or git on master we need to skip this test.

Feel free to do that on the master server (e.g., one can modify the script or alias git). The need for a local workaround does not, IMO, justify violating an important principle in the official script.

> If you want to make it run in a robust way ...

Personally, I do not want that -- I think a git upgrade is an overall better way forward here. However, I would not block quality attempts to compensate for the lack of `git --only-matching` support (in dedicated PRs).

Reply:
IMO this does not "admit MakeNamedErrorDetail bugs" because jenkins and semaphore build checks run before each commit use a version of git which copes fine with that option. It is only on the master machine performing release maintenance where it breaks the automation.

I am happy to move the script change to its own PR for clarity if you like. I only left it in the maintenance brance for this PR because I knew this would be getting a QA review before merge, and as expected you spotted it.

PS. the v4 and v5 updates went in smoothly because they were 100% automatic changes. This maintainer commit was not needed for those versions.

Reply:
FYI: `--only-matching` became available since git [2.19](https://git-scm.com/docs/git-grep/2.19.0#git-grep--o).

Reply:
> IMO this does not "admit MakeNamedErrorDetail bugs" because jenkins and semaphore build checks run before each commit use a version of git which copes fine with that option

1. AFAIK, Semaphore checks do not run `scripts/source-maintenance.sh`. That is still a (complicated) TODO. Not sure about Jenkins, but I suspect the same is true there.
2. Jenkins and Semaphore do not insist that git they use supports `--only-matching` so even if they start running `scripts/source-maintenance.sh`, they might admit MakeNamedErrorDetail bugs in the future if the proposed script change is accepted.

> I am happy to move the script change to its own PR for clarity if you like.

Extracting a PR-unrelated problem into its own PR will not address that problem, of course, but it will help move _this_ PR forward. AFAICT, this change request will apply "as is" to the new PR (containing the same modifications).

> I only left it in the maintenance brance for this PR because I knew this would be getting a QA review before merge, and as expected you spotted it.

FYI: Spotting this problem in a supposed-to-be-automated-by-existing-scripts modification of 2000+ files was an accident.